File: PieGallery/app/src/main/java/com/deadely/piegallery/base/BaseActivity.java
package com.deadely.piegallery.base;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.deadely.piegallery.App;
import com.deadely.piegallery.di.component.ActivityComponent;
import com.deadely.piegallery.di.module.ActivityModule;

public abstract class BaseActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityComponent activityComponent = App.getInstance().getComponent().activityComponent(new ActivityModule(this));
        inject(activityComponent);
    }

    protected abstract void inject(ActivityComponent activityComponent);
}

File: itlen/app/src/main/java/com/deadely/itl_en/base/BaseActivity.kt
package com.deadely.itl_en.base

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.deadely.itl_en.App
import com.deadely.itl_en.base.mvp.IBaseProgressView
import com.deadely.itl_en.di.component.ActivityComponent
import com.deadely.itl_en.di.module.ActivityModule

abstract class BaseActivity : AppCompatActivity(), IBaseProgressView {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val activityComponent = App.instance.component.activityComponent(ActivityModule(this))
        inject(activityComponent!!)
    }

    protected abstract fun inject(activityComponent: ActivityComponent)

    override fun startLoading() {
    }

    override fun completeLoading() {
    }

    override fun errorLoading() {
    }
}